### PR TITLE
fix: #1101

### LINKED ISSUE #1101: getDisplayMedia() does not work on MacOS Sonoma 14.0+
**Describe the bug**
If you try getting the user's screen capture it fails with a `NotAllowedError`, this is also an issue reported in tauri, https://github.com/tauri-apps/tauri/issues/2600

**Steps To Reproduce**
- Change the simple.rs example to display any webpage that has a getDisplayMedia call (Gmeet, etc)
- Run `cargo run --example simple`
- Try to share screen from the video calling app
- It will fail

**Expected behavior**
Ideally, the display media should be captured and shown on the screen, just like camera and audio work out of the box

**Screenshots**
Copied the screen shot from the same issue
image

**Platform and Versions (please complete the following information):**
OS: macOS 14.0 (23A344)
Rustc: 1.72.1

**Would you want to assign yourself to resolve this bug?**
- [ ] Yes
- [x] No

### PR DESCRIPTION
**Pull request by tarushnagpal**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### Checklist
- [x] This PR will resolve #1101
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary
- [x] It can be built on all targets and pass CI/CD.

### CODE REVIEW

File: src/wkwebview/mod.rs [hunk 1]
```diff
@@ -1170,6 +1174,22 @@ pub fn platform_webview_version() -> Result<String> {
...
+        Ok(webview_system_and_major_version[..1].to_string())
+      } else {
+        Ok(webview_system_and_major_version[..2].to_string())
+      }
+    }
+    Err(e) => {
+      return Err(e);
+    }
+  }
+}
```
Comment (pewsheen) [inline src/wkwebview/mod.rs]:
```suggestion
pub fn platform_webview_system_version() -> Result<String> {
  platform_webview_version().map(|webview_version| {
    let webview_system_and_major_version = webview_version.split('.').next().unwrap();
    if webview_system_and_major_version.chars().count() < 5 {
      webview_system_and_major_version[..1].to_string()
    } else {
      webview_system_and_major_version[..2].to_string()
    }
  })
}
```

Some clean up, also make clippy happy

File: src/wkwebview/mod.rs [hunk 2]
```diff
@@ -787,11 +787,15 @@ impl InnerWebView {
...
           );
 
-          // Disable media dialogs
-          ctl.add_method(
-            sel!(webView:requestMediaCapturePermissionForOrigin:initiatedByFrame:type:decisionHandler:),
-            request_media_capture_permission as extern "C" fn(&Object, Sel, id, id, id, id, id),
-          );
+          // Only disable media dialogs on macOS < 14.0
+          // https://tauri.app/v1/references/webview-versions/
+          if !platform_webview_system_version()?.parse::<i32>().is_ok_and(|version| version > 18) {
```
Comment (pewsheen) [inline src/wkwebview/mod.rs]:
```suggestion
          let webview_system_version = platform_webview_system_version()?.parse::<i32>();
          if webview_system_version.is_err() || webview_system_version.unwrap() < 19 {
```

Could be more readable?

**Comment by pewsheen** [comment]
Oh, you need to [sign your commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) so we can merge them.

File: src/wkwebview/mod.rs (hunk 1 above)
Comment (tarushnagpal) [reply src/wkwebview/mod.rs]: Thanks, still very new to rust!

**Review by pewsheen — APPROVED** [review]

**Comment by tarushnagpal** [comment]
Apparently signing a previous commit isn't possible without squashing them all, just made a new PR here https://github.com/tauri-apps/wry/pull/1111, sorry for the trouble

**Pull request closed without merging**